package/index.d.ts

```diff
@@ -0,0 +1,53 @@
+declare module "teaful" {
+
+  import React from "react";
+  
+  type HookReturn<T> = [T, (value: T) => void, () => void];
+  type initialStoreType = Record<string, any>;
+
+  type Hook<S> = (
+    initial?: S,
+    onAfterUpdate?: afterCallbackType<S>
+  ) => HookReturn<S>;
+
+  type HookDry<S> = (initial?: S) => HookReturn<S>;
+
+  export type Hoc<S> = { store: HookReturn<S> };
+
+  type HocFunc<S, R extends React.ComponentClass = React.ComponentClass> = (
```

Comment:
wondering if we should type the R as
`
R extends React.ComponentType 
`
as the ComponentType would cover both  React.FunctionComponent<> | React.ClassComponent<>

Reply:
Our advice is to use the `useStore` hook rather than the `withStore` HOC whenever it is a functional component. So I don't think it's wrong to have only the `React.ClassComponent`. In which cases would it make sense to use the `withStore` in a functional component?

If there is some sense that we didn't take into account then I think it's okay to change it. 🤔